### ISSUE TITLE
Heroic instance lock not enforced.

### ISSUE THREAD

**Issue by netherspite** [netherspite/cc-buglist#831]
<a href="https://github.com/r00ty-tc"><img src="https://avatars1.githubusercontent.com/u/173349?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [r00ty-tc](https://github.com/r00ty-tc)**    
_Sunday Jul 05, 2015 at 01:27 GMT_  
_Originally opened as https://github.com/ccshiro/cc-buglist/issues/831_

----

There's actually a second aspect to this. But, the issues may be linked.

First issue. Upon re-logging my player that logged out in group in a heroic group (Shadow lab) that had killed the first boss. I found that the entire instance had re spawned around me.

How it should work, once locked to an instance, the bosses killed should not be available to kill. Of course trash leading up to that boss should not spawn. The heroic lock was still showing in my raid window.

Second issue. When I relogged I was in the same position I logged out at. Now, if the group is still formed and the instance is the same one (at the same stage, then this is correct behavior).

Expected result: If the instance resets while logged out, when logging in, you appear at the inside of the instance just inside the entrance (the spawn point as you enter the portal).

**Comment by netherspite** [comment]
<a href="https://github.com/ccshiro"><img src="https://avatars2.githubusercontent.com/u/8135895?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ccshiro](https://github.com/ccshiro)**  
_Tuesday Jul 14, 2015 at 18:31 GMT_

----

In regards to your first issue: In heroics killing a boss does not prevent trash leading up to it to respawn. That mechanic exists in raids in TBC, but not in heroics. The main point here is whether or not you actually verified the boss was there again (in which case it's a bug), or if it wasn't (in which case it's working as intended).

Second issue depends on the first issue; if it's the same instance (which it might be), the behavior is correct.

Tl;dr: Did you actually check that the boss was there again?

**Comment by netherspite** [comment]
<a href="https://github.com/r00ty-tc"><img src="https://avatars1.githubusercontent.com/u/173349?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [r00ty-tc](https://github.com/r00ty-tc)**  
_Tuesday Jul 14, 2015 at 18:35 GMT_

----

It's not possible to check if the boss is there in Shadow lab. He spawns after you kill the trash sadly. Otherwise I would have checked. I was fairly sure the trash didn't respawn in heroics either. But, I'll admit to not really doing heroic instances in multiple parts. So, no real evidence on that one.

**Comment by netherspite** [comment]
<a href="https://github.com/ccshiro"><img src="https://avatars2.githubusercontent.com/u/8135895?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ccshiro](https://github.com/ccshiro)**  
_Sunday Feb 14, 2016 at 12:51 GMT_

----

Possibly related to #2249.
